src/algebra/gcd_domain.lean

```diff
@@ -521,6 +521,37 @@ nat.dvd_antisymm (by unfold gcd; exact nat.gcd_dvd_left _ _)
 theorem gcd_eq_right {i j : ℤ} (H : j ∣ i) : gcd i j = nat_abs j :=
 by rw [gcd_comm, gcd_eq_left H]
 
+theorem ne_zero_of_gcd {x y : ℤ}
+  (hc : gcd x y ≠ 0) : x ≠ 0 ∨ y ≠ 0 :=
+begin
+  by_cases hx : x = 0,
+  { apply or.intro_right, intro hy, revert hc, rw [hx, hy], simp },
+  tauto
+end
+
+theorem exists_gcd_one {m n : ℤ} (H : 0 < gcd m n) :
+  ∃ (m' n' : ℤ), gcd m' n' = 1 ∧ m = m' * gcd m n ∧ n = n' * gcd m n :=
+⟨_, _, gcd_div_gcd_div_gcd H,
+  (int.div_mul_cancel (gcd_dvd_left m n)).symm,
+  (int.div_mul_cancel (gcd_dvd_right m n)).symm⟩
+
+theorem exists_gcd_one' {m n : ℤ} (H : 0 < gcd m n) :
+  ∃ (g : ℕ), ∃ (m' n' : ℤ), 0 < g ∧ gcd m' n' = 1 ∧ m = m' * g ∧ n = n' * g :=
+let ⟨m', n', h⟩ := exists_gcd_one H in ⟨_, m', n', H, h⟩
+
+theorem pow_dvd_pow_of_dvd {a b : ℤ} (h : a ∣ b) : ∀ n:ℕ, a^n ∣ b^n
```

Comment:
This doesn't belong in this file, please move it to `algebra/ring` and prove it for an arbitrary `comm_semiring`.

Reply:
I placed this in algebra/group_power, because algebra/ring doesn't understand pow...

Reply:
Ooh, right. It should be done for `comm_monoid`s... I certainly didn't generalise far enough in my other suggestion.